modules/imgcodecs/perf/perf_png.cpp

```diff
@@ -58,6 +58,51 @@ PERF_TEST(PNG, encode)
     SANITY_CHECK_NOTHING();
 }
 
+const int PNGFilterFlags[] = {
+       IMWRITE_PNG_NO_FILTERS,
+       IMWRITE_PNG_FILTER_NONE,
+       IMWRITE_PNG_FILTER_SUB,
+       IMWRITE_PNG_FILTER_UP,
+       IMWRITE_PNG_FILTER_AVG,
+       IMWRITE_PNG_FILTER_PAETH,
+       IMWRITE_PNG_FAST_FILTERS,
+       IMWRITE_PNG_ALL_FILTERS
+};
+
+PERF_TEST_P(PNG, params_filter, testing::ValuesIn(PNGFilterFlags))
+{
+    String filename = getDataPath("perf/1920x1080.png");
+
+    Mat src = imread(filename);
+    EXPECT_FALSE(src.empty()) << "Cannot open test image perf/1920x1080.png";
+    vector<uchar> buf;
+
+    TEST_CYCLE() imencode(".png", src, buf, { IMWRITE_PNG_FILTER, GetParam() });
+
+    std::cout << "Encoded buffer size: " << buf.size()
+        << " bytes, Compression ratio: " << std::fixed << std::setprecision(2)
+        << (static_cast<double>(buf.size()) / (src.total() * src.channels())) * 100.0 << "%" << std::endl;
+
+    SANITY_CHECK_NOTHING();
+}
+
+PERF_TEST_P(PNG, DISABLED_params_filter_max_compression, testing::ValuesIn(PNGFilterFlags))
```

Comment:
Why is it disabled?

Reply:
when a common user run it takes so long time ( --perf_force_samples  default value is 100 )